cumulusci/tasks/bulkdata.py

```diff
@@ -602,7 +678,62 @@ def _import_results(self, mapping, result_file, conn):
             process_incoming_rows(result_file, record_type), self.logger
         )
         data_file = IteratorBytesIO(processor)
-        self._sql_bulk_insert_from_csv(conn, mapping["table"], columns, data_file)
+        if mapping["oid_as_pk"]:
+            self._sql_bulk_insert_from_csv(conn, mapping["table"], columns, data_file)
+        else:
```

Comment:
There's a fair amount happening below; can you add some comments and docstrings? (I could improve on that in my work in this module too.)